### PR TITLE
[TS] LPS-169318 Reverting usage of DLFileVersion.userId back to ResourcePermission.ownerId in dslQuery due to performance issues

### PR DESCRIPTION
**Pull request by evibudai**
Hi Team,

https://issues.liferay.com/browse/LPS-169318

Third time's the charm.

Previous PRs:
https://github.com/liferay-core-infra/liferay-portal/pull/2697: caused regression because  of failing test
https://github.com/liferay-core-infra/liferay-portal/pull/2798: was advised that the first solution would be better

@yuhai 's comments:
1) I think userIdColumn check is not necessary in InlineSQLHelperImpl class. It should still be ResourcePermissionTable.INSTANCE.ownerId
2) Your first fix https://github.com/liferay-core-infra/liferay-portal/pull/2697 looks good to me. Please firstly try to fix the test. 

In my first PR I replaced DLFileVersion.userId with ResourcePermission.ownerId, and I fixed the tests.

Best regards,
Évi

### CODE REVIEW

**Comment by tinatian** [comment]
@evibudai In your commit message, you said "Reverting userId to ownerId", then why we made change to use owerId before?

**Comment by evibudai** [comment]
hi @tinatian , I don't know why they made that change, but as we agreed with @yuhai on the previous PR, this is the correct way to formulate this sql query.

File: modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java [hunk 1]
```diff
@@ -418,10 +418,8 @@ private <T extends Table<T>> Predicate _getPermissionPredicate(
 
 		T table = classPKColumn.getTable();
 
-		Column<T, Long> userIdColumn = table.getColumn("userId", Long.class);
-
 		DSLQuery resourcePermissionDSLQuery = _getResourcePermissionQuery(
-			permissionChecker, modelClassName, userIdColumn, groupIds);
+			permissionChecker, modelClassName, groupIds);
```
Comment (yuhai) [inline modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java:422]: @evibudai, please place **T table = classPKColumn.getTable();** to https://github.com/liferay-core-infra/liferay-portal/blob/58592900a29f25e693ab25adc86e70e12adabc35/modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java#L467.

Comment (evibudai) [reply modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java:420]: @yuhai it's done

**Comment by yuhai** [comment]
> @evibudai In your commit message, you said "Reverting userId to ownerId", then why we made change to use owerId before?

@tinatian, as far as I can find the beginning history is LPS-8869 7ddb2a25469ce014daabac8da89646ae9d38bae7, it used userIdField. And then LPS-16843 2de86e1446e4175754f1869c051b63d25585abc7 added ResourcePermission.ownerId.

For the current portal, I checked its usage InlineSQLHelper#getPermissionWherePredicate(),  other usages own userId column in own table. Except ObjectEntryLocalServiceImpl class,  if its usages also own useId column, that it to say, ResourcePermission.ownerId doesn't been used. 

From my understanding, for models permission, when creating one model entry, it should equal using ResourcePermission.ownerId or table.userId  when ResourcePermission.scope = 4 (https://github.com/liferay-core-infra/liferay-portal/blob/58592900a29f25e693ab25adc86e70e12adabc35/modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java#L498-L522).

@evibudai, I think we also need to check https://github.com/liferay-core-infra/liferay-portal/blob/58592900a29f25e693ab25adc86e70e12adabc35/modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/InlineSQLHelperImpl.java#L560-L581. This place existed similar usage.

**Comment by yuhai** [comment]
> @evibudai In your commit message, you said "Reverting userId to ownerId", then why we made change to use owerId before?



> @evibudai In your commit message, you said "Reverting userId to ownerId", then why we made change to use owerId before?

@tinatian, sorry, my reply should be incorrect for the question. My reply should be for InlineSQLHelperImpl#_getResourcePermissionSQL().

The case is from DLFileEntryFinderImpl#_getJoinStep. LPS-136334(https://github.com/liferay-core-infra/liferay-portal/commit/61b5aee358cfc4748a50b0112b63a95562ded72c) changed this. Before the fix, it used DLFileEntryFinderImpl#getFileEntriesSQL() -> InlineSQLHelperUtil.replacePermissionCheck(String sql, String className, String classPKField, long groupId) to generate the sql. The api implied used userIdField=null as param. Finally, it will invoke InlineSQLHelperImpl#_getResourcePermissionSQL() and then use owerId.

For the current case, since it used InlineSQLHelperUtil.getPermissionWherePredicate(), so the previous ownId was replaced by userId. And this api was added by LPS-106397 https://github.com/liferay-core-infra/liferay-portal/commit/98cb017537f452839f15f777b58ec9733091e1d1. userIdColumn and ownId were created together. I guess Preston might refer to InlineSQLHelperImpl#_getResourcePermissionSQL() to write the part logic.

For the current userIdField usage in InlineSQLHelperImpl, it looks weird since InlineSQLHelperImpl#replacePermissionCheck() usages look use userIdField is null. I will further check this.

**Comment by yuhai** [comment]
resend it to https://github.com/liferay-core-infra/liferay-portal/pull/2989

**Pull request closed without merging**